livre/eq/Config.cpp

```diff
@@ -60,6 +97,65 @@ class Config::Impl
         , dataFrameRange( INVALID_FRAME_RANGE )
     {}
 
+    void gatherHistogram( const Histogram& histogram, const float area, const uint32_t currentId )
+    {
+        const uint32_t oldestId = histogramQueue.empty() ? -1u :
+                                  histogramQueue.back().id;
+
+        // If we get a very old frame skip it
+        if( oldestId != -1u && currentId < oldestId )
+            return;
+
+        const ViewHistogram viewHistogram( histogram, area, currentId );
+        auto it = histogramQueue.begin();
+        while( it != histogramQueue.end( ))
```

Comment:
You don't need a while loop, just checking the first is enough. If you look carefully at your code you will notice that after that at the end of first iteration dataMerged can't be anything but true (due to the return statement above).